MoreLinq/ExceptBy.cs

```diff
@@ -24,6 +24,54 @@ namespace MoreLinq
     static partial class MoreEnumerable
     {
         /// <summary>
+        /// Returns a distinct set of elements from the first collection,
+        /// whose keys are not in the second collection.
+        /// </summary>
+        /// <typeparam name="T">The type of source, keys, and result elements.</typeparam>
+        /// <param name="first">The source collection.</param>
+        /// <param name="second">The key collection.</param>
+        /// <param name="keySelector">The key selector.</param>
+        /// <returns>
+        /// Distinct set of elements from first collection,
+        /// whose keys are not in second collection.
+        /// </returns>
+        public static IEnumerable<T> ExceptBy<T>(this IEnumerable<T> first,
+            IEnumerable<T> second,
+            Func<T, T> keySelector) {
```

Comment:
Is this key selector really necessary? Doesn't look like it to me. Would removing this paramtere make the type inference problems go away?